File: BACKEND/TM.Infrastructure/Repositories/TaskItemRepository.cs
using Microsoft.EntityFrameworkCore;
using TM.Domain.Entity;
using TM.Domain.Enum;
using TM.Domain.Interface;
using TM.Infrastructure.Context.TMContext;

namespace TM.Infrastructure.Repositories.TaskItemRepository;

public class TaskItemRepository : ITaskItemRepository
{
    private readonly TMContext _context;

    public TaskItemRepository(TMContext context)
    {
        _context = context;
    }

    public async Task<bool> CreateTaskItem(TaskItem taskItem, CancellationToken cancellationToken)
    {
        bool response = false;
        if(taskItem != null)
        {
            taskItem.CreatedDate = DateTime.UtcNow;
            taskItem.UpdatedDate = DateTime.UtcNow;
            await _context.TaskItem.AddAsync(taskItem, cancellationToken);
            response = await _context.SaveChangesAsync(cancellationToken) > 0;   
        }
        return response;
    }

    public async Task<List<TaskItem>> GetAllTaskItem(CancellationToken cancellationToken)
    {
        return await _context.TaskItem.Where(t => t.Status != TaskStatusEnum.Deleted).ToListAsync(cancellationToken);
    }

    public async Task<TaskItem> GetTaskItemById(int id, CancellationToken cancellationToken)
    {
        var response = await _context.TaskItem.FirstOrDefaultAsync(t => t.Id == id, cancellationToken);
        return response != null ? response : new TaskItem();
    }

    public async Task<bool> RemoveTaskItem(int id, CancellationToken cancellationToken)
    {
        bool response = false;
        if(id > 0)
        {
            var currentTask = await _context.TaskItem.FirstOrDefaultAsync(t => t.Id == id, cancellationToken);
            if(currentTask != null)
            {
                _context.TaskItem.Remove(currentTask);
                response = await _context.SaveChangesAsync(cancellationToken) > 0;
            }
        }
        return response;
    }

    public async Task<bool> UpdateTaskItem(TaskItem taskItem, CancellationToken cancellationToken)
    {
        bool response = false;
        if(taskItem != null)
        {
            taskItem.UpdatedDate = DateTime.UtcNow;
            _context.TaskItem.Update(taskItem);
            response = await _context.SaveChangesAsync(cancellationToken) > 0;
        }
        return response;
    }
}

File: FRONTEND/TM.Frontend/src/infrastructure/repository/taskItemRepository.ts
import { HttpStatusCode } from "axios";
import type { TaskItem } from "../../domain/entity/TaskItem";
import type { ITaskItemRepository } from "../../domain/interface/ITaskItemRepository";
import { httpClient } from "../api/httpClient";
import { mapGetTaskToTaskItem, mapTaskToCreateTaskRequest, mapTaskToUpdateTaskRequest } from '../mappers/taskItemMapper';
import type { CreateTaskInput } from "../../application/task/createTaskInput";
import type { UpdateTaskInput } from "../../application/task/updateTaskInput";


export class taskItemRepository implements ITaskItemRepository{

    async createTaskItem(taskItem: CreateTaskInput): Promise<boolean> {
        const dto = mapTaskToCreateTaskRequest(taskItem);
        const response = await httpClient.post("TaskItem/Create", dto);
        return response.status == HttpStatusCode.Ok ? true : false; 
    }

    async updateTaskItem(taskItem: UpdateTaskInput): Promise<boolean> {
        const dto = mapTaskToUpdateTaskRequest(taskItem);
        const response = await httpClient.put("TaskItem/Update", dto);
        return response.status == HttpStatusCode.Ok ? true : false;
    }

    async removeTaskitem(id: number): Promise<void> {
        await httpClient.put("TaskItem/Remove", {Id: id});
    }

    async getTaskItemById(id: number): Promise<TaskItem> {
        const response =  await httpClient.put("TaskItem/GetById", {Id: id})
        const task = mapGetTaskToTaskItem(response.data);
        return task;
    }

    async getAllTaskItem(): Promise<TaskItem[]> {
        const response = await httpClient.get("TaskItem/GetAll");
        
        if (!response.data || !Array.isArray(response.data)) {
            return [];
        }

        const tasks: TaskItem[] = response.data.map((item: any) => {
            return mapGetTaskToTaskItem(item);
        });
        
        return tasks;
    }

}